### PR TITLE
[DOC] Describe how to propose a migration script to the community.

### PR DESCRIPTION
**Pull request by legalsylvain**
(git command, commit name convention, etc.)

CC : @remytms, @etobella, @bguillot 

@pedrobaeza : I'm sure you have an opinion on contribution conventions. ;-)

### CODE REVIEW

File: docsource/090_contribute.rst [hunk 1]
```diff
@@ -10,6 +10,45 @@ In order to contribute to the OpenUpgrade project, please
...
+How to contribute new migration scripts
+---------------------------------------
+
+To Contribute to OpenUpgrade you must make one pull request per module.
+
+For example, if you want to propose the migration script of the ``account``
+module from version 15.0 to version 16.0:
+
+  * Always ensure that there is no work already in progress by a community member.
+    For that purpose, go the issue named "Migration to version 16.0" on github
```
Comment (pedrobaeza) [inline docsource/090_contribute.rst]:
```suggestion
    For that purpose, go the issue named "Migration to version NN.0" on GitHub, where NN is the version for which you want to propose the script.
```

File: docsource/090_contribute.rst [hunk 2]
```diff
@@ -10,6 +10,45 @@ In order to contribute to the OpenUpgrade project, please
...
+    `here <https://github.com/OCA/OpenUpgrade/issues/3681/>`_.
```
Comment (pedrobaeza) [inline docsource/090_contribute.rst]:
```suggestion
    `here <https://github.com/OCA/OpenUpgrade/issues/3681/>`_.
    
    Sometimes, the pull request is not yet registered in the issue, so it's not bad to do a search on the opened pull requests with the name of the module to look for.
```

File: docsource/090_contribute.rst [hunk 3]
```diff
@@ -10,6 +10,45 @@ In order to contribute to the OpenUpgrade project, please
...
+
+  * Then, ensure that all the dependencies of the module are mark as ``done`` or
```
Comment (pedrobaeza) [inline docsource/090_contribute.rst:27]: This is not a requirement for developing the script. Just to merge it.

File: docsource/090_contribute.rst [hunk 4]
```diff
@@ -10,6 +10,45 @@ In order to contribute to the OpenUpgrade project, please
...
+    ``Nothing to do`` in OpenUpgrade for that version.
+    For that purpose, refer to :doc:`030_coverage_analysis`.
+    In our example, check the ``depends`` key of the `manifest <https://github.com/odoo/odoo/blob/16.0/addons/account/__manifest__.py#L18>`_ of the ``account`` module 
+    If some dependencies are missing, you should start by migrating these modules.
+
+  * As other OCA contribution, create a new branch, from an up to date OCA branch:
+
+    ``git checkout -b 16.0-mig-account``
+
+  * Make a copy of the analysis file.
```
Comment (pedrobaeza) [inline docsource/090_contribute.rst]:
```suggestion
  * Make a copy of the analysis file with the suffix ``_work``.
```

File: docsource/090_contribute.rst [hunk 5]
```diff
@@ -10,6 +10,45 @@ In order to contribute to the OpenUpgrade project, please
...
+
+    ``cd ./openupgrade_scripts/scripts/account/16.0.1.2/``
+    ``cp upgrade_analysis.txt upgrade_analysis_work.txt``
+
+  * Explain the changes to do in the ``upgrade_analysis_work.txt``.
+    Write ``pre-migration.py`` and / or ``post-migration.py`` scripts in the same folder.
```
Comment (pedrobaeza) [inline docsource/090_contribute.rst]: Put this as new points instead of the same one.

File: docsource/090_contribute.rst [hunk 6]
```diff
@@ -10,6 +10,45 @@ In order to contribute to the OpenUpgrade project, please
...
+  * As other OCA contribution, create a new branch, from an up to date OCA branch:
+
+    ``git checkout -b 16.0-mig-account``
+
+  * Make a copy of the analysis file.
+
+    ``cd ./openupgrade_scripts/scripts/account/16.0.1.2/``
+    ``cp upgrade_analysis.txt upgrade_analysis_work.txt``
+
+  * Explain the changes to do in the ``upgrade_analysis_work.txt``.
```
Comment (pedrobaeza) [inline docsource/090_contribute.rst:53]: Explain here about the format of the comments:

- `NOTHING TO DO`, and why it's nothing to do
- `DONE: pre-migration/post-migration:` indicating the what and maybe the why.
- `TODO:` This shouldn't be usually done.

Then, also explain that they can reorder lines and group them together for including all of them in a "logical" block for putting only one comment including all of them. Examples: all the noupdate=0 new and del ir* records, or the `NEW` and `DEL` lines for a field that is renamed.

File: docsource/090_contribute.rst [hunk 7]
```diff
@@ -10,6 +10,45 @@ In order to contribute to the OpenUpgrade project, please
...
+    Write ``pre-migration.py`` and / or ``post-migration.py`` scripts in the same folder.
+    Comment / uncomment lines in ``noupdate_changes.xml``.
+    (Read more in :doc:`080_migration_script_development`)
+
+  * Finally, commit your changes.
+    ``git add . && git commit -am "[OU-ADD] account"``
+    (For a fix, use ``[OU-FIX]`` prefix)
```
Comment (pedrobaeza) [inline docsource/090_contribute.rst]: Put the possible tags in a separate line explaining it: `[OU-ADD]`, `[OU-FIX]`, `[OU-IMP]`. We should propose these tags also to https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#71commit-message

File: docsource/090_contribute.rst [hunk 8]
```diff
@@ -10,6 +10,45 @@ In order to contribute to the OpenUpgrade project, please
...
+
+  * Propose your changes to the community for review.
+    ``git push MY_REMOTE 16.0-mig-account``
```
Comment (pedrobaeza) [inline docsource/090_contribute.rst]: and open the pull request.

**Comment by legalsylvain** [comment]
Hi @pedrobaeza. Thanks for your suggestions ! 

1) I made some changes. Could you take a look on the last commit ? (I'll squash all commits, once all is OK)

2) I'm just not sure about that remark : https://github.com/OCA/OpenUpgrade/pull/4227#discussion_r1389034201
I mean : 
-  to make a Pull request, it is indeed not mandatory
- to make tests (or use CI), if dependancies are not done, there is a risk that the tests will fail for unrelated reason. 

I think it's a good practice to contribute, following the dependency order.

What do you think ?

I can add a text like : 

``
It's not strictly necessary to make contributions in the order of dependency, but you may face errors that are not related to your contribution, if you don't follow this order. For example, the sale_stock migration script may fail, if the stock migration script has not been run upstream.
```


Thanks !

**Comment by legalsylvain** [comment]
Friendly reminder on this PR.

Thanks for your review.

**Review by etobella — APPROVED** [review]

**Review by bguillot — APPROVED** [review]

**Comment by bguillot** [comment]
Thank you Sylvain.

**Comment by pedrobaeza** [comment]
@legalsylvain OK about your text proposal about the pull requests.

**Comment by legalsylvain** [comment]
> @legalsylvain OK about your text proposal about the pull requests.

done.

File: docsource/090_contribute.rst [hunk 9]
```diff
@@ -10,6 +10,85 @@ In order to contribute to the OpenUpgrade project, please
...
+---------------------------------------
+
+To Contribute to OpenUpgrade you must make one pull request per module.
+
+For example, if you want to propose the migration script of the ``account``
+module from version 15.0 to version 16.0:
+
+  * Always ensure that there is no work already in progress by a community member.
+    For that purpose, go the issue named "Migration to version NN.0" on GitHub, where NN is the version for which you want to propose the script.
+    `here <https://github.com/OCA/OpenUpgrade/issues/3681/>`_.
```
Comment (pedrobaeza) [inline docsource/090_contribute.rst]: Are we going to keep this issue reference? It won't go well through time, when we were on Odoo 20 (or just already in 17).

Comment (legalsylvain) [reply docsource/090_contribute.rst]: Well, the sentence before is "For example, if you want to propose the migration script of the ``account``
module from version 15.0 to version 16.0" so it make sense.
but well, replaced by "https://github.com/OCA/OpenUpgrade/issues?q=is%3Aopen++is%3Aissue+%22Migration+to+version%22+"

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**